### ISSUE TITLE
Inconsistency: action.setIcon() & action.setBadgeBackgroundColor()

### ISSUE THREAD

**Issue by erosman** [hober/web-extensions#336]
There are some minor inconsistencies. 
_N.B. Only Firefox & Chrome checked._

#### Firefox
[browserAction.setIcon()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/setIcon) & [action.setIcon()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/action/setIcon)

> If each one of `imageData` and `path` is one of `undefined`, `null` or empty object:
> 
> - If `tabId` is specified, and the tab has a tab-specific icon set, then the tab will inherit the icon from the window to which it belongs.
> - If `windowId` is specified, and the window has a window-specific icon set, then the window will inherit the global icon.
> - Otherwise, the global icon will be reset to the manifest icon.

```js
browser.action.setIcon({});
```

#### Chrome
[setIcon](https://developer.chrome.com/docs/extensions/reference/action/#method-setIcon)

> Either the **path** or the **imageData** property must be specified.

```js
chrome.action.setIcon({});
// Error: Either the path or imageData property must be specified.
```

#### Firefox
[browserAction.setBadgeBackgroundColor()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/setBadgeBackgroundColor) & [action.setBadgeBackgroundColor()](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/action/setBadgeBackgroundColor)

> `null`. If a `tabId` is specified, it removes the tab-specific badge background color so that the tab inherits the global badge background color. Otherwise it reverts the global badge background color to the default value.

```js
browser.action.setBadgeBackgroundColor({color: null});
```

### Chrome
[setBadgeBackgroundColor](https://developer.chrome.com/docs/extensions/reference/action/#method-setBadgeBackgroundColor)

**Resetting to the default value**
```js
chrome.action.setBadgeBackgroundColor({color: null});
// Error in invocation of action.setBadgeBackgroundColor(object details, optional function callback): Error at parameter 'details': Missing required property 'color'.
```

#### Firefox
```js
browser.action.setBadgeBackgroundColor({color: 'transparent'});
```

#### Chrome
```js
chrome.action.setBadgeBackgroundColor({color: 'transparent'});
// Error: The color specification could not be parsed.
```

---
#### Footnote

See also: #216

**Comment by hanguokai** [comment]
Yes, `action.setBadgeTextColor()` also has this minor inconsistency. Since it can be worked around, it is generally not a problem.

**Comment by xeenon** [comment]
Safari supports null resetting the action properties. We also plan to support `windowId`.

**Comment by carlosjeurissen** [comment]
So basically the inconsistency is Chrome does not allow resetting the properties? Potentially we can just upgrade this issue to propose allowing to reset the action properties.

@oliverdunk is this something Chrome would want to implement? Seems an open issue has been made already: https://bugs.chromium.org/p/chromium/issues/detail?id=1488303

> Safari support null resetting the action properties.

@xeenon Wonderful! Is this something recently implemented?

**Comment by xeenon** [comment]
Safari has always supported null reseting, IIRC.

**Comment by zombie** [comment]
Similar to Safari, I think this was supported from the get go.

**Comment by Rob--W** [comment]
Devlin expressed support for resetting to the default. The crbug to track is at https://issues.chromium.org/issues/40073862

**Comment by carlosjeurissen** [comment]
The same inconsistency applies to `action.setPopup`. Setting the popup to null resets it in Firefox, while in Chrome it throws an exception.

@xeenon Just tested setting popup to null in Safari 17.4.1 and contrary to what is stated on MDN, it behaves as if it is set to an empty string.
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/setPopup

**Comment by xeenon** [comment]
@carlosjeurissen Hmm thanks. What about Safari Technology Preview?

**Comment by carlosjeurissen** [comment]
@xeenon Seems it has the correct behaviour in the Technology Preview Release 194 (Safari 17.4, WebKit 19619.1.11.111.2)

**Comment by xeenon** [comment]
@carlosjeurissen Thanks for checking! I'll make a note to update MDN.
